pkg/proc/proc.go

```diff
@@ -58,6 +60,42 @@ func HostPath(p string) string {
 	return path.Join(Path, strconv.Itoa(1), p)
 }
 
+func (p *Proc) FindCGroupPathForPID(pid PID) (string, error) {
+	cgroupData, err := p.procFS.ReadFile(fmt.Sprintf("%d/cgroup", pid))
+	if err != nil {
+		return "", err
+	}
+
+	var emptyFallback string
+
+	for _, line := range strings.Split(string(cgroupData), "\n") {
+		// Last line will be empty, we simply ignore it.
+		if len(line) == 0 {
```

Comment:
I think you wouldn't have to handle this case if `strings.Fields(string(cgroupData))` was used in the range loop header instead.

Reply:
I do not fully understand what you mean. `string.Fields` splits by whitespace, not by newline 🤔